tools/loader/ImageClassifier.cpp

```diff
@@ -430,12 +464,15 @@ int main(int argc, char **argv) {
   llvm::outs() << "Model: " << loader.getFunction()->getName() << "\n";
 
   int numErrors = 0;
-  while ((streamInputFilenamesMode &&
-          getNextImageFilenames(&inputImageBatchFilenames)) ||
-         (miniBatchMode &&
-          getNextMiniBatch(inputImageBatchFilenames, inputImageFilenames,
-                           minibatchIndex, miniBatch)) ||
-         isFirstRun) {
+
+  while (!iterationsOpt &&
```

Comment:
After looking through this it looks to me like there's a lot of shared logic between here inside the while loop, and in the `iterationsOpt` case after this while loop.

I'm wondering if either you could merge most of the `iterationsOpt` logic into the while loop here, or otherwise break out of the while loop early before we run inference so that many of the data structures are correctly set up? E.g. this loop sets up `inputImageBatchFilenames`, compiles the model, etc. which you also do below.

Reply:
Yeah it's been a few iterations, I took another look at moved some logic in, but not all I think this is a pretty clean approach.